### PR TITLE
Update cstrike module - 1

### PR DESCRIPTION
**Pull request by Arkshine**
This is based on #254 .

This patch is a logic step after implementing a game config system to avoid any hardcoding of game datas. While I was on it, I've fixed/improved/cleanup the obvious things.

Essentially:
* All game datas are now external and since we can dump all  the classes headers of the binary, this includes as well offsets of most of classes associated to an entity.
* An approach on-demand of hooks used by some natives/forwards. In others words, if these natives/forwards are not used, module does nothing.
* Some hostages natives support now CZ
* Some natives gain an extra param
* Worth to be noted: `cs_set/reset_user_model`, `cs_set_user_team` won't trigger anymore an overflow resulting a crash when used numerous times in a row. This is an engine bug, and the new workaround in this patch is basically postponing the update of models to the next frame.
* Some minor optimizations

That's all for now.

### CODE REVIEW

**Comment by xPaw** [comment]
Just a nit, replace spaces with underscores in links.
e.g: `Gamedata_Updating_(AMX Mod X)` :arrow_right: `Gamedata_Updating_(AMX_Mod_X)`

**Comment by Arkshine** [comment]
Making suddenly such message be hookable by plugins can lead to some unexpected behaviors. So probably not a good idea unless we add a new param or something, but doing that for the related natives seem like silly.

File: plugins/include/cstrike.inc [hunk 1]
```diff
@@ -127,13 +127,14 @@ enum
 #define CSI_SG552               CSW_SG552
 #define CSI_AK47                CSW_AK47
 #define CSI_P90                 CSW_P90
-#define CSI_SHIELDGUN           CSW_SHIELDGUN
+#define CSI_SHIELDGUN           CSW_SHIELDGUN   // The real CS value, use CSI_SHELD instead.
```
Comment (Freeman-AM) [inline plugins/include/cstrike.inc]: Typo

**Comment by Arkshine** [comment]
Tested all the related natives and it worked fine. Unless I did not test properly or missed something, this should be fine.

**Comment by Nextra** [comment]
Good that you tested again, should cover the big issues. This is really hard to go through in detail in a reasonable time frame. We can probably :ship:

**Pull request merged**